nemo/collections/asr/parts/submodules/aed_decoding/aed_batched_streaming.py

```diff
@@ -0,0 +1,562 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from dataclasses import dataclass
+
+import torch
+from omegaconf import OmegaConf
+
+from nemo.collections.asr.models.aed_multitask_models import EncDecMultiTaskModel, parse_multitask_prompt
+from nemo.collections.asr.parts.submodules.multitask_decoding import AEDStreamingDecodingConfig
+from nemo.collections.asr.parts.utils.streaming_utils import ContextSize
+from nemo.utils import logging
+
+
+@dataclass
+class AEDStreamingState:
+    decoder_input_ids: torch.Tensor = None  # tokens ids of initial AED model prompt
+    pred_tokens_ids: torch.Tensor = None  # buffer with predicted tokens ids
+    decoding_step: int = -1  # current decoding step
+    decoder_mems_list: list = None  # decoder caches, helps to reduce the memory usage
+    is_last_chunk_batch: torch.Tensor = False  # whether the current chunk is the last speech chunk in the audio
+    max_generation_length: int = (
+        256  # maximum number of tokens to be generated for each sample (can be bigger for long audio)
+    )
+    max_tokens_per_one_second: int = 10  # maximum number of tokens to be generated per one second of audio
+    max_tokens_per_alignatt_step: int = (
+        None  # maximum number of tokens to be generated for each step of alignatt decoding policy
+    )
+    use_avgpool_for_alignatt: bool = False  # use avgpooling for alignatt decoding policy
+    tokens_frame_alignment: torch.Tensor = (
+        None  # frame alignment of the predicted tokens (used for LAAL calculation in alignatt)
+    )
+    prev_encoder_shift: int = 0  # previous encoder shift (used for LAAL calculation in alignatt)
+    device: torch.device = None
+
+
+class GreedyBatchedStreamingAEDComputer(ABC):
```

Comment:
Is inheritance from `ABC` needed? 
It looks like there are no abstract methods in this class -> ABC is unnecessary